cmd/util/cmd/epochs/cmd/recover.go

```diff
@@ -0,0 +1,248 @@
+// EpochRecover event to be emitted. This command retrieves the current protocol state identities, computes the cluster assignment using those
+// identities, generates the cluster QC's and retrieves the DKG key vector of the last successful epoch.
+// This recovery process has some constraints:
+//   - The RecoveryEpoch must have exactly the same consensus committee as participated in the most recent successful DKG.
+//   - The RecoveryEpoch must contain enough "internal" collection nodes so that all clusters contain a supermajority of "internal" collection nodes (same constraint as sporks)
+var (
+	generateRecoverEpochTxArgsCmd = &cobra.Command{
+		Use:   "efm-recover-tx-args",
+		Short: "Generates recover epoch transaction arguments",
+		Long: `
+Generates transaction arguments for the epoch recovery transaction.
+The epoch recovery transaction is used to recover from any failure in the epoch transition process without requiring a spork.
+This recovery process has some constraints:
+- The RecoveryEpoch must have exactly the same consensus committee as participated in the most recent successful DKG.
+- The RecoveryEpoch must contain enough "internal" collection nodes so that all clusters contain a supermajority of "internal" collection nodes (same constraint as sporks)
+`,
+		Run: generateRecoverEpochTxArgs(getSnapshot),
+	}
+
+	flagAnAddress                string
+	flagAnPubkey                 string
+	flagInternalNodePrivInfoDir  string
+	flagNodeConfigJson           string
+	flagCollectionClusters       int
+	flagNumViewsInEpoch          uint64
+	flagNumViewsInStakingAuction uint64
+	flagEpochCounter             uint64
+)
+
+func init() {
+	rootCmd.AddCommand(generateRecoverEpochTxArgsCmd)
+	err := addGenerateRecoverEpochTxArgsCmdFlags()
+	if err != nil {
+		panic(err)
+	}
+}
+
+func addGenerateRecoverEpochTxArgsCmdFlags() error {
+	generateRecoverEpochTxArgsCmd.Flags().IntVar(&flagCollectionClusters, "collection-clusters", 3,
+		"number of collection clusters")
+	// required parameters for network configuration and generation of root node identities
+	generateRecoverEpochTxArgsCmd.Flags().StringVar(&flagNodeConfigJson, "config", "",
+		"path to a JSON file containing multiple node configurations (fields Role, Address, Weight)")
+	generateRecoverEpochTxArgsCmd.Flags().StringVar(&flagInternalNodePrivInfoDir, "internal-priv-dir", "", "path to directory "+
+		"containing the output from the `keygen` command for internal nodes")
+	generateRecoverEpochTxArgsCmd.Flags().Uint64Var(&flagNumViewsInEpoch, "epoch-length", 4000, "length of each epoch measured in views")
+	generateRecoverEpochTxArgsCmd.Flags().Uint64Var(&flagNumViewsInStakingAuction, "epoch-staking-phase-length", 100, "length of the epoch staking phase measured in views")
+	generateRecoverEpochTxArgsCmd.Flags().Uint64Var(&flagEpochCounter, "epoch-counter", 0, "the epoch counter used to generate the root cluster block")
```

Comment:
Based on my understanding, I would think these default values are drastically to short. would also like to get @jordanschalm's thoughts ...

My thoughts:
* Default values should provide a sensible setting for mainnet. 
* We are talking about an epoch here, with the limitation that it is mainly for recovery, _but_ aren't we planning to run a regular DKG in this recovery epoch? I think a day would be reasonable 

   * For reference, we decided on 3000 views for each DKG phase (3 in total) for the mainnet consensus timing, plus `EpochCommitSafetyThreshold` (which should also be _minimally_ 1000 blocks), plus staking and Epoch commit phase that also belong to an epoch on the happy path. 
     With the new consensus timing [👉   [reference](https://www.notion.so/flowfoundation/Cruise-Control-headroom-for-speedups-46dc17e07ae14462b03341e4432a907d?pvs=4#4d398f482125429f9460bccdceebdc47)], one day would correspond to to 108,000 views for mainnet

Reply:
In practice the CLI will require all of these to be explicitly specified in flags when run, because of the `MarkFlagRequired` blocks below. I think it is preferable to force operators to provide specific values and fail rather than using a default value (current behaviour). 

To prevent confusion, we could set the "default" value parameter when defining the flag to 0.